src/base/bittorrent/tracker.cpp

```diff
@@ -357,7 +404,11 @@ void Tracker::processAnnounceRequest()
         unregisterPeer(announceReq);
     }
     else {
-        throw TrackerError("Invalid \"event\" parameter");
+        LogMsg(tr(R"(Embedded Tracker: Invalid "event" parameter, assuming "empty")"
+                R"( in request from peer: %1, request port: %2)")
+                .arg(announceReq.socketAddress.toString(), announceReq.requestPort)
+            , Log::WARNING);
+        announceReq.event = ANNOUNCE_REQUEST_EVENT_EMPTY;
```

Comment:
why? I don't think this make sense.

Reply:
Again, why is more leniency here not desirable? A warning will still be logged.

Reply:
I don't think you really answered my question...

Reply:
>Invalid "event" parameter, assuming "empty"

Is this behavior a part of protocol standard? If no I strongly object to encourage invalid clients! All invalid requests should be dropped.

Reply:
@glassez BEP 3 does not specify what should be done in the case of an invalid value for this parameter (i.e. it does not explicitly say "a request containing any other values should/must be considered invalid"). So I think it is better to be lenient, and just log a warning.Otherwise, we can just keep the logging statement explaining that the request was dropped.

Reply:
@arvidn Can we get your opinion on this matter? In this case, and also the `numwant` discussion below, is it better to be lenient and attempt serve the request if possible, falling back to a sane default, or to drop the request altogether?

Reply:
without having thought about this case very much, generally I think it's a good default to be strict. (imagine if all bittorrent clients had required bencoded dictionaries to be sorted correctly for instance, clients wouldn't have to go out of their way to support invalid encodings as they do today).

Maybe some relevant questions for this case. (1) a client that sends an invalid request to the tracker, is it likely to be a correct about other arguments, say the info-hash? (2) is it the kind of peer that a tracker should hand out to ther peers?

Reply:
@arvidn I see.
1, 2 - Probably not, on both counts. So I guess being strict is the better option.

But in any case, shouldn't the spec be updated to be more strict? At least the V2 spec, as updating the original one is not meaningful at this point. If the spec is not strict enough, it allows implementations to make assumptions that could lead to ossification of the protocol - the most well-known case of this is TLS 1.x -> 1.3 I think. Isn't it desirable to prevent that?